project/npda/views/patient.py

```diff
@@ -240,6 +244,12 @@ def get_context_data(self, **kwargs):
     def form_valid(self, form: BaseForm) -> HttpResponse:
         patient = form.save(commit=False)
         patient.is_valid = True
+
+        if patient.postcode:
+            imd = imd_for_postcode(patient.postcode)
+            if imd:
```

Comment:
Sneaky bug fix, don't overwrite an IMD score if the API transiently returns an error